### ISSUE TITLE
Build initial Paperless reports

### ISSUE THREAD

**Issue by jasonperrone** [bdeterling/axiom#14]
1. Move Types by Date
2. Move Types by Hour for a date
3. Move Types by User
4. No screen shot, but the menu implies Move Types by User by Hour for a date and presumably a specific user.

The first two and the last one seem to a fit a time-series graph with time on the axis, and a series for each move type. The Ad Hoc report would almost work, but we need to add support for multiple series.

The third one probably needs to be a table because presumably there are a lot of users so a graph would get too busy. If we add a new type of table that let you choose two columns, sort of like a graph does, then we could use that.

**Comment by bdeterling** [comment]
Right, that’s why these two tasks are on the list:

	• Add multi-series graph option
	• Add cross table? (table that shows all possible values from a field as columns)

**Comment by jasonperrone** [comment]
LOL. You're answering yourself. All I did was cut and paste your email into the case.

**Comment by bdeterling** [comment]
At least I’m consistent!

**Comment by jasonperrone** [comment]
Brain storming the by-hour part. We could do something like detect when you have picked a Date field and then present a select where you can select a date format, and we group on dates based on that format. So, if you pick Date, you can then pick dd/mm/yyyy hh:mm, signifying you want it grouped down to the hour, whereas dd/mm/yyyy signifies you want it grouped to the day.

Or we put options on the Axis parameters additional to Date like month, hour, week, minute. 

Not sure which is the best way so let me know if you already had some idea how you'd like it done.

I do believe that once you take care of my PR, reports 1 and 3 are done. I do realize you had an idea for #1 that involved a bar chart and drilldowns.  You can currently make the bar chart you asked for, but the drilldowns are a problem:

1) There are drilldowns now that go to a table, but they are not filtering correctly
2) The drilldown you asked for about going to a line chart is beyond by bandwidth for this week.

In summary, I need you to:

1) Decide how we handle the display configuration UI for the hourly (and other time period) breakouts and let me know and I'll implement it
2) Fix the drilldowns that already exist on the bar chart for moves by type.

**Comment by jasonperrone** [comment]
Assigning to Brian to address my questions above and to refine my tasks.

**Comment by bdeterling** [comment]
This is what I'm thinking for drilldowns. It's ambitious, but will be cool.

* A new optional method in a report def available_drilldowns(report_configuration)
* That will return an array of classes that will handle the drilldown based on the passed in configuration
* On the graph, if there are 0 drlldowns, then clicking does nothing, one drilldown and clicking will just call it directly, more than one an it will display a context menu letting them pick where the drilldown. 
* Possibly we can let the save their choice in the configuration so they don't have to click every time.
* The drilldown report will be like a normal report but will take an extra filter for the drilldown x and y point/value.
* So a bar graph might have drilldowns for List (table showing items), Time-series, and Hour of Day. A time series could have List and Zoom where zoom shows more detail (not sure about that).

Thoughts?

**Comment by jasonperrone** [comment]
These "classes". Are they also functional as standalone reports?

**Comment by bdeterling** [comment]
Probably not, but they could be. Would rather use modules to share common code and have separate classes otherwise it’s pretty confusing.
